x-pack/plugins/transform/public/app/common/validators/string_validator.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { requiredErrorMessage, stringNotValidErrorMessage } from './messages';
+
+import type { Validator } from './types';
+
+export const stringValidator: Validator = (value, isOptional = true) => {
```

Comment:
we already have a generic [requiredValidator](https://github.com/elastic/kibana/blob/8453fe820a7bd0b264b3963f547a5092b5e811d5/x-pack/plugins/ml/common/util/validators.ts#L61)